artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/Page.java

```diff
@@ -120,105 +110,133 @@ public void setLiveCache(LivePageCache pageCache) {
          throw ActiveMQMessageBundle.BUNDLE.invalidPageIO();
       }
 
-      final List<PagedMessage> messages = new ArrayList<>();
-
       size.lazySet((int) file.size());
 
-      if (this.canBeMapped) {
-         readFromMapped(storage, messages);
```

Comment:
The original readFromSequentialFile() allocates file size memory and may cause long TTSP. 
But readFromMapped() only calls mmap64() and gets starting address of the mapping which should not cause long TTSP. Why we not reserve mmap file read here?

Reply:
The long TTSP issue of the mapped version wasn't caused during the cheap mmap64, but while reading the MappedByteBuffer, on each new OS page that wasn't mapped onto the OS page cache.
The original version were having other issues eg memory footprint...but not long TTSP.

Reply:
Got it, thanks for the elaboration :)

Reply:
I forgot to answer you on

> Good job. By the way, How did you find the issue?

Using clustering we have measured such long time to safepoint pauses that was causing the broker to become unresponsive so I've investigated on them using an approach similar to https://www.mail-archive.com/mechanical-sympathy@googlegroups.com/msg01122.html.
I have found that `MappedByteBuffer::get` was making the CPU to get descheduled for several seconds, making the whole JVM to freeze. 
The filesystem used was XFS so the reason behind such behaviour wasn't dependent by any ext4 journalling daemon, but due to major page faults + I/O activity to fill the OS page cache.
As I've said: assuming infinite RAM and the fastest disk ever with XFS, this issue is very unlikely to happen...